js/src/components/google-combo-account-card/claim-ads-account/claim-ads-account.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Flex, FlexBlock } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import ClaimAdsAccountButton from './claim-ads-account-button';
+import Section from '.~/wcdl/section';
+import useWindowFocus from '.~/hooks/useWindowFocus';
+import useGoogleAdsAccountStatus from '.~/hooks/useGoogleAdsAccountStatus';
+import { useAppDispatch } from '.~/data';
+import './claim-ads-account.scss';
+import useUpsertAdsAccount from '.~/hooks/useUpsertAdsAccount';
+
+/**
+ * ClaimAdsAccount component.
+ *
+ * @return {JSX.Element} ClaimAdsAccount component.
+ */
+const ClaimAdsAccount = () => {
+	const [ , forceUpdate ] = useState( 0 );
+	const isWindowFocused = useWindowFocus();
+	const { hasAccess, hasFinishedResolution, inviteLink, step } =
+		useGoogleAdsAccountStatus();
+	const [ upsertAdsAccount, { loading } ] = useUpsertAdsAccount();
+	const { invalidateResolution } = useAppDispatch();
+	const claimButtonClickedRef = useRef( false );
+	const claimInProgressRef = useRef( false );
+
+	useEffect( () => {
+		if ( isWindowFocused && claimButtonClickedRef.current ) {
+			invalidateResolution( 'getGoogleAdsAccountStatus' );
```

Comment:
@ankitrox There won't be the need to invalidate the data when we'll use the suggested approach above, i.e using `useWindowFocusCallbackIntervalEffect`